pkg/operation/botanist/controlplane.go

```diff
@@ -300,6 +293,46 @@ func (b *Botanist) HibernateControlPlane(ctx context.Context) error {
 	return nil
 }
 
+// ScaleControlPlaneDeploymentsToZero scales the kube-apiserver, kube-controller-manager and gardener-resource-manager to zero
+func (b *Botanist) ScaleControlPlaneDeploymentsToZero(ctx context.Context) error {
+	if err := b.scaleKubernetesDeployments(ctx, 0, b.K8sSeedClient.Client()); err != nil {
+		return err
+	}
+
+	if err := b.K8sSeedClient.Client().Delete(ctx, &hvpav1alpha1.Hvpa{ObjectMeta: metav1.ObjectMeta{Name: v1beta1constants.DeploymentNameKubeAPIServer, Namespace: b.Shoot.SeedNamespace}}, kubernetes.DefaultDeleteOptions...); err != nil {
+		if !apierrors.IsNotFound(err) && !meta.IsNoMatchError(err) {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// ScaleETCDStatefulSetsToZero scales ETCD main and events to zero
+func (b *Botanist) ScaleETCDStatefulSetsToZero(ctx context.Context) error {
+	for _, statefulset := range []string{v1beta1constants.ETCDEvents, v1beta1constants.ETCDMain} {
+		if err := kubernetes.ScaleStatefulSet(ctx, b.K8sSeedClient.Client(), kutil.Key(b.Shoot.SeedNamespace, statefulset), 0); client.IgnoreNotFound(err) != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (b *Botanist) scaleKubernetesDeployments(ctx context.Context, replicas int32, c client.Client) error {
```

Comment:
```suggestion
func (b *Botanist) scaleControlPlaneDeploymentsToZero(ctx context.Context, replicas int32, c client.Client) error {
```

Reply:
Can we not specify zero, because you can set whatever replica count you want. 
Or we can remove the parameter and hardcode it to zero. wdyt

Reply:
Sure, lets make the replica count configurable and rename the function.

Reply:
I also changed my mind... actually it could not be reused because if the deployments are missing we are ignoring the error. In case where we scale up i think this error deserves to be propageted.